components/layout/display_list_builder.rs

```diff
@@ -1953,6 +1953,12 @@ impl BlockFlowDisplayListBuilding for BlockFlow {
         // we don't want it to be clipped by its own scroll root.
         let containing_scroll_root_id = self.setup_scroll_root_for_block(state);
 
+        match self.positioning() {
+            position::T::absolute | position::T::relative | position::T::fixed =>
```

Comment:
Perhaps a new method that returns a boolean if this matches could avoid some duplication?